model-optimizer/extensions/load/tf/loader.py

```diff
@@ -100,5 +101,64 @@ def load(self, graph: Graph):
 
         graph.check_empty_graph('protobuf2nx. It may happen due to problems with loaded model')
         extract_node_attrs(graph, lambda node: tf_op_extractor(node, check_for_duplicates(tf_op_extractors)))
+
+        graph.graph['layout'] = 'NCHW' if argv.disable_nhwc_to_nchw else 'NHWC'
+
+        # try to detect layout from the nodes of the graph. If there are no convolution nodes in N(D)HWC layout then we
+        # consider that the graph is in NCHW layout and no layout conversion should be performed
+        if not argv.force_nhwc_to_nchw:
+            NHWC_conv_detected = graph_or_sub_graph_has_nhwc_ops(graph)
+            if not NHWC_conv_detected:
+                if not argv.silent and not argv.disable_nhwc_to_nchw:
+                    print('The TensorFlow model does not contain Convolution operations with N(D)HWC layout so the '
+                          'model layout conversion from NHWC to NCHW has been disabled (the same effect as if you '
+                          'specify "--disable_nhwc_to_nchw" command line parameter). If you still want to enable the '
+                          'layout conversion mechanism add "--force_nhwc_to_nchw" command line parameter.')
+                for_graph_and_each_sub_graph_recursively(graph, update_cmd_params_and_layout)
+
         send_op_names_info(framework, graph)
         send_shapes_info(framework, graph)
+
+
+def is_node_layout_nhwc(node: Node):
+    """
+    Check the layout attribute of specific operations and return True if any of them has layout NHWC.
+    :param node: Node to check
+    :return: Boolean result of the check
+    """
+    if node.soft_get('op') in ["Conv2D", "DepthwiseConv2dNative", "Conv3D", "Conv2DBackpropInput",
+                               "Conv3DBackpropInputV2"]:
+        if node.soft_get('layout') in ["NHWC", "NDHWC"]:
+            log.debug('Detected convolution node with NHWC layout: "{}"'.format(node.soft_get('name', node.id)))
+            return True
+    return False
+
+
+def graph_or_sub_graph_has_nhwc_ops(graph: Graph):
+    """
+    Checks that a graph or any sub-graph (inside Loop) operation contains nodes with NHWC layout.
+    :param graph: main graph to check
+    :return: Boolean result of the check
+    """
+    NHWC_conv_detected = False
+    for node in graph.get_op_nodes():
+        if is_node_layout_nhwc(node):
+            NHWC_conv_detected = True
+            break
+
+        # for the Loop node we need to check that the body does not contain marker ops as well
+        if node.op == 'Loop':
```

Comment:
use node.sofr_get('op')

Reply:
we need add check for If operation? if yes, please add comment about if.

Reply:
'op' attribute always exists, so there is no need for `soft_get`

Reply:
Comment will be added with the next code update